pkg/operator/ceph/controller/spec.go

```diff
@@ -84,12 +97,19 @@ fi
 # edit the logrotate file to only rotate a specific daemon log
 # otherwise we will logrotate log files without reloading certain daemons
 # this might happen when multiple daemons run on the same machine
-sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FILE"
+sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FOLDER"
 
 while true; do
-	sleep "$PERIODICITY"
-	echo "starting log rotation"
-	logrotate --verbose --force "$LOG_ROTATE_CEPH_FILE"
+	sleep "30m"
+	CURRENT_TIME_IN_SEC=$(date +%s)
+	TIME_DIFF_IN_HOUR=$(echo $((CURRENT_TIME_IN_SEC - START_TIME_IN_SEC)) | awk '{printf "%d", $1/3600'})
+	if [[ $TIME_DIFF_IN_HOUR -ge $PERIODICITY ]] || [[ $SIZE_OF_FILE -ge $LOG_MAX_SIZE ]]; then
+		echo "starting log rotation"
+		logrotate --verbose --force "$LOG_ROTATE_CEPH_FOLDER"
+		START_TIME_IN_SEC=$(date +%s)
+		SIZE_OF_FILE=$(du -sh $LOG_ROTATE_CEPH_FOLDER | awk '{print $1}' | sed 's/.$//')
```

Comment:
why do we ned the size of the file again? It was already found above, right?
```suggestion
		SIZE_OF_FILE=$(du -sh $LOG_ROTATE_CEPH_FOLDER/$CEPH_CLIENT_ID.log | awk '{print $1}' | sed 's/.$//')
```